mempool/clist_mempool.go

```diff
@@ -92,8 +87,6 @@ func NewCListMempool(
 		mp.cache = NopTxCache{}
 	}
 
-	proxyAppConn.SetResponseCallback(mp.globalCb)
```

Comment:
The `globalCb` callback was only used for Recheck responses. Now the response is simply handled by a callback in the `ReqRes` structure.

Reply:
There are three structures with a callback:
- `proxyAppConn`, implementing the `Client` interface, with a callback of type `func(*types.Request, *types.Response)`. Before this PR, this callback was used only on CheckTx ABCI responses. It is set only once, during the creation of `CListMempool`.
- `ReqRes`, with a callback of type `func(*types.Response)`. This is more flexible because we can set it when `CheckTxAsync` returns the request/response, allowing us to customise the function by passing extra information, such as the sender.
- the external callback passed to `CListMempool.CheckTx`, reintroduced in this PR, and invoked at the end of handling the CheckTx response. This is used for processing the response on the RPC endpoints `broadcast_tx_sync` and `broadcast_tx_commit`.

Reply:
> The globalCb callback was only used for Recheck responses.

I see there a `case` for `CheckTx` in addition to the one for `ReCheckTx`

Reply:
Should this useful explanation be somewhere in the form of a comment?